src/pymor/models/interface.py

```diff
@@ -109,6 +109,74 @@ def _compute_output(self, solution, mu=None, **kwargs):
         else:
             return self.output_functional.apply(solution, mu=mu).to_numpy()
 
+    def _compute_solution_d_mu_single_direction(self, parameter, index, solution, mu=None, **kwargs):
+        """Compute the partial derivative of the solution w.r.t. a parameter index
+
+        Parameters
+        ----------
+        parameter
+            parameter for which to compute the sensitivity
+        index
+            parameter index for which to compute the sensitivity
+        solution
+            Internal model state for the given |Parameter value|.
+        mu
+            |Parameter value| for which to solve
+
+        Returns
+        -------
+        The sensitivity of the solution as a |VectorArray|.
+        """
+        raise NotImplementedError
+
+    def _compute_solution_d_mu(self, solution, mu=None, **kwargs):
+        """Compute all partial derivative of the solution w.r.t. a parameter index
+
+        Parameters
+        ----------
+        solution
+            Internal model state for the given |Parameter value|.
+        mu
+            |Parameter value| for which to solve
+
+        Returns
+        -------
+        A dict of all partial sensitivities of the solution.
```

Comment:
Would it maybe make more sense to return a single VectorArray, given that we also don't return a dict for `output_d_mu`?

Reply:
You are right that this does not go along with the way `output_d_mu` is returned. However, IMO internally it is way easier to store the sensitivities as a dict and I also find this more user friendly if one needs to play around with the sensitivities. With a dict the sensitivities are accessible without knowing the index in the VectorArray, this `component` and `index` corresponds to. This is also used in `_compute_output_d_mu`. Of course, one could also change the way `_compute_output_d_mu` uses this, but I like it more like this.